### PR TITLE
WCS 1.1.1 support

### LINKED ISSUE #114: Missing <ows:Title> and <ows:Abstract> elements in WCS GetCapabilities response
Those layer descriptors are optional but are already defined in layers description inside the service configuration and would help the layer discovery in GIS tools such as QGIS. 
```
layers = {
   hrdtm = {
        type = "simplesourceconf"
        name = "hrdtm"
        title = "High Resolution Digital Terain Model (HRDTM)"
        abstract = "High Resolution Digital Terain Model generated from LiDAR"
        source = {
            ...

```
With <ows:Title> and <ows:Abstract> missing, layers appear as unamed in QGIS menu
capture

### LINKED ISSUE #115: Layer definition elements unused in WMS GetCapabilities response
Layer definition elements (title, abstract) are not properly rendered in WMS GetCapabilities response.

Following layer definition
```
layers = {
   hrdtm = {
        type = "simplesourceconf"
        name = "hrdtm"
        title = "High Resolution Digital Terain Model (HRDTM)"
        abstract = "High Resolution Digital Terain Model generated from LiDAR"
        source = {
            ...
```
will produce the following output.
```
<Layer queryable="false">
   <Name>hrdtm</Name>
   <Title>hrdtm</Title>
   <Abstract>hrdtm</Abstract>
...
```
Faulty code seems to be: https://github.com/geotrellis/geotrellis-server/blob/564ba5e846ab7d89ef65ff9ea7f865c5c261c86d/ogc/src/main/scala/geotrellis/server/ogc/wms/CapabilitiesView.scala#L105-L107

### LINKED ISSUE #142: Bad assembly strategy
Right now the [build.sbt file has a not fully correct assembly strategy](https://github.com/geotrellis/geotrellis-server/blob/develop/build.sbt#L56-L62), it should be replaced by 
```scala
assemblyMergeStrategy in assembly := {
  case "reference.conf"   => MergeStrategy.concat
  case "application.conf" => MergeStrategy.concat
  case PathList("META-INF", xs @ _*) =>
    xs match {
      case ("MANIFEST.MF" :: Nil) =>
        MergeStrategy.discard
      case ("services" :: _ :: Nil) =>
        MergeStrategy.concat
      case ("javax.media.jai.registryFile.jai" :: Nil) | ("registryFile.jai" :: Nil) | ("registryFile.jaiext" :: Nil) =>
        MergeStrategy.concat
      case (name :: Nil) if name.endsWith(".RSA") || name.endsWith(".DSA") || name.endsWith(".SF") =>
        MergeStrategy.discard
      case _ =>
        MergeStrategy.first
      }
  case _ => MergeStrategy.first
}
```

Since the current version can confuse users, and doesn't take into account [GeoTrellis SPI merge](https://github.com/geotrellis/geotrellis-spark-job.g8/blob/427c022eb0c169f5c65018fafe78ce2e88a80cd2/src/main/g8/build.sbt#L51).

### LINKED ISSUE #188: Generate WCS 1.1.1 protocol using XSD data model
The current protocol version is too limited as it is hand written. 

The task it so generate WCS 1.1.1 protocol from the OGC XSD schema. This would be done towards the future temporal support.

### LINKED ISSUE #192: WCS 1.1.1 GetCoverage Support
At this point the current master branch in fact supports only WCS 1.0.0 [width & heght params are only required for v1.0.0](https://github.com/geotrellis/geotrellis-server/blob/develop/ogc/src/main/scala/geotrellis/server/ogc/wcs/params/WCSParams.scala#L27-L28). 

For the WCS 1.1.1 the requests look like:

```
?SERVICE=WCS&
VERSION=1.1.1&
REQUEST=GetCoverage&
FORMAT=image/tiff;subtype%3D"geotiff"&
IDENTIFIER=DEM&
BOUNDINGBOX=614337.3125,122998.92500000000291038,614987.6875,123176.07499999999708962,urn:ogc:def:crs:EPSG::900913&
GRIDBASECRS=urn:ogc:def:crs:EPSG::900913&
GRIDCS=urn:ogc:def:cs:OGC:0.0:Grid2dSquareCS&
GRIDTYPE=urn:ogc:def:method:WCS:1.1:2dSimpleGrid&
GRIDORIGIN=614337.3125,123176.07499999999708962&GRIDOFFSETS=130.0749999999922295,-88.57500000000193552
```

The task is to implement WCS 1.1.1 `GetCoverage` requests support.

### PR DESCRIPTION
**Pull request by pomadchin**
## Overview

This PR adds WCS 1.1.1 protocol generation and a basic support. 
WCS 1.0.0 is no longer supported.

### Checklist

- [x] Description of PR is in an appropriate section of the CHANGELOG and grouped with similar changes if possible

Closes #188
Closes #192
Closes #114
Closes #115
Closes #142

### CODE REVIEW

**Comment by pomadchin** [comment]
This PR also contains a backported fix https://github.com/locationtech/geotrellis/issues/3179 that should be removed after the new GeoTrellis version release.

File: build.sbt [hunk 1]
```diff
@@ -212,22 +223,23 @@ lazy val opengis = project
...
+    Compile / scalaxb / scalaxbDispatchVersion := dispatchVer,
+    Compile / scalaxb / scalaxbPackageName := "generated",
+    Compile / scalaxb / scalaxbProtocolPackageName := Some("opengis"),
+    Compile / scalaxb /  scalaxbPackageNames := Map(
+      uri("http://www.w3.org/1999/xlink")           -> "xlink",
+      uri("http://www.opengis.net/wms")             -> "opengis.wms",
+      uri("http://www.opengis.net/ogc")             -> "opengis.ogc",
+      uri("http://www.opengis.net/wmts/1.0")        -> "opengis.wmts",
+      uri("http://www.opengis.net/ows/1.1")         -> "opengis.ows",
+      uri("http://www.opengis.net/ows")             -> "opengis.sld.ows",
```
Comment (pomadchin) [inline build.sbt]: This PR fixes `ows 1.0` packages location \cc @jisantuc @notthatbreezy

File: core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+import geotrellis.proj4._
+import geotrellis.raster.io.geotiff.{Auto, AutoHigherResolution, Base, OverviewStrategy}
+import geotrellis.raster.resample.{NearestNeighbor, ResampleMethod}
+import geotrellis.raster._
+import geotrellis.layer._
+import geotrellis.vector._
+
+/**
+  * TODO: remove after upgrading up to GeoTrellis 3.3.0
```
Comment (CloudNiner) [inline core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala]: Do we have an issue in this project to upgrade to GT 3+? If not, lets create one. And once we have an issue, can you note in it that we should be removing these legacy classes?

Comment (pomadchin) [reply core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala]: @CloudNiner this project is already updated up to the latest GT; this one porbably to upgrade exactly up to the next GT release https://github.com/locationtech/geotrellis/pull/3180

Comment (pomadchin) [reply core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala]: https://github.com/geotrellis/geotrellis-server/issues/197

**Pull request merged**